capstone/src/test/java/com/google/step/servlets/RestaurantServletTest.java

```diff
@@ -0,0 +1,187 @@
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    /**
+     * Test for the doGet method of the RestaurantServlet. First adds
+     * a dummy entity to the mock Datastore then calls doGet using a
+     * ServletRunner to check the response against the expected format.
+     * @throws IOException
+     * @throws SAXException if the ServletRunner parser cannot handle the request
+     */
+    @Test
+    public void testDoGet() throws IOException, SAXException {
+        DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+        // Create dummy RestaurantInfo entity and add it to the mock Datastore
+        Entity restaurantInfo1 = new Entity("RestaurantInfo");
+        restaurantInfo1.setProperty("restaurantKey", 4);
+        restaurantInfo1.setProperty("name", "Wildfire");
+        restaurantInfo1.setProperty("location", new GeoPt((float) 42.17844, (float) -87.92843));
+        restaurantInfo1.setProperty(
+                "story", "Swanky American chain serving steak, chops & seafood, plus burgers, sides & cocktails.");
+        List<String> cuisineList1 = Arrays.asList("Steakhouse", "American");
+        restaurantInfo1.setProperty("cuisine", cuisineList1);
+        restaurantInfo1.setProperty("phone", "(847)279-7900");
+        restaurantInfo1.setProperty("website",
+                "https://www.wildfirerestaurant.com");
+        restaurantInfo1.setProperty("score", 1.1);
+        restaurantInfo1.setProperty("status", "STRUGGLING");
+        datastoreService.put(restaurantInfo1);
+
+        // Set expected JSON response based on props
+        String expected = "{\"restaurant\":\"{\\\"value\\\":{\\\"restaurantKey\\\":4,\\\"name\\\":\\\"Wildfire\\\"," +
+                "\\\"location\\\":{\\\"latitude\\\":42.17844,\\\"longitude\\\":-87.92843},\\\"story\\\":\\\"" +
+                "Swanky American chain serving steak, chops \\\\u0026 seafood, plus burgers, sides \\\\u0026 cocktails.\\\"," +
+                "\\\"cuisine\\\":[\\\"Steakhouse\\\",\\\"American\\\"],\\\"phone\\\":\\\"(847)279-7900\\\",\\\"website\\\":\\\"" +
+                "https://www.wildfirerestaurant.com\\\",\\\"status\\\":\\\"STRUGGLING\\\"}}\"}\n";
+
+        // Create a client to run the servlet
+        ServletRunner sr = new ServletRunner();
+        sr.registerServlet("restaurant", RestaurantServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // This can throw the SAXException.
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/restaurant?restaurantKey=4");
+        WebResponse response = sc.getResponse(request);
+        assertNotNull("No response received", response);
+        assertEquals("content type", "application/json", response.getContentType());
+        String actual = response.getText();
+
+        assertEquals(expected, actual);
```

Comment:
In cases like this, I like to use the assertEquals(description, expected, actual) method, since for example both parameters are a variable with a not very descriptive name, the description at the beginning helps understand what's the comparison.

Reply:
Martin makes a good point. What I typically like to do instead is use descriptive variable names for the `expected` and `actual` parameters, but either way works, I think.